### PR TITLE
feat(Accordion)!: Decoupled AccordionArea and AccordionButton

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
### PR Checklist

- ~[ ] Related to Abstract designs:~
- [x] Related to JIRA ticket: WEB-955
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### Description

Some consumers of the now-deprecated Accordion in the monolith use it as a controlled component, such as in the Get Help menu where only one section may be expanded at a time. I played around with allowing the `expanded` prop to be optional but the test surface was exponentially larger and more complex.

~_(making draft PR of changes now so I don't lose them while recreating my borked repo at home... and I need to make lunch)_~ ✅

Consumers can now use uncontrolled `AccordionArea` and `AccordionButton` components and will need to supply their own `expanded` state, like so:

```tsx
const [expanded, setExpanded] = useState(expanded);

return (
  <AccordionArea
    expanded={expanded}
    top={
      <AccordionButton
        expanded={expanded}
        onClick={() => setExpanded(!expanded)}
      >
        Click Me
      </AccordionButton>
    }
  >
    (contents)
  </Accordion>
);
```

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.52-alpha.5fcfa1.0<br/>@codecademy/gamut@12.1.1-alpha.5fcfa1.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f1b1cddb5e249a99a753d55--codecademy-styleguide.netlify.app>https://5f1b1cddb5e249a99a753d55--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f1b1cddb5e249a99a753d55>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.52-alpha.30b201.0<br/>@codecademy/gamut@12.1.1-alpha.30b201.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f1ef284e57de7048d1d9957--codecademy-styleguide.netlify.app>https://5f1ef284e57de7048d1d9957--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f1ef284e57de7048d1d9957>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.7.52-alpha.93c33a.0<br/>@codecademy/gamut@12.1.1-alpha.93c33a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f1ef6960f291b9173f0e062--codecademy-styleguide.netlify.app>https://5f1ef6960f291b9173f0e062--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5f1ef6960f291b9173f0e062>Deploy Logs</a></p>

File: packages/styleguide/stories/About/constants.tsx [hunk 1]
```diff
@@ -25,7 +25,7 @@ export const ABOUT_SECTIONS = [
     emoji: '🧬',
     description:
       'Molecules are groups of atoms together. These consist of a few atoms combined for a small, self-contained design.',
-    examples: ['Alert', 'Accordion'],
+    examples: ['Accordion', 'Alert'],
```
Comment (codecaaron) [inline packages/styleguide/stories/About/constants.tsx:28]: ALPHABETIC!

Comment (JoshuaKGoldberg) [reply packages/styleguide/stories/About/constants.tsx:28]: It was hurting me

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.8.1-alpha.c30297.0<br/>@codecademy/gamut@12.3.0-alpha.c30297.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f207ee2b5e24e01b5ec992b--gamut-preview.netlify.app>https://5f207ee2b5e24e01b5ec992b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f207ee2b5e24e01b5ec992b>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.8.1-alpha.9dddcf.0<br/>@codecademy/gamut@12.3.0-alpha.9dddcf.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f219ee1e7c1ae01aaea70af--gamut-preview.netlify.app>https://5f219ee1e7c1ae01aaea70af--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f219ee1e7c1ae01aaea70af>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.8.4-alpha.b2bdc4.0<br/>@codecademy/gamut@12.2.3-alpha.b2bdc4.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f21a9db4b154c01d1170d4c--gamut-preview.netlify.app>https://5f21a9db4b154c01d1170d4c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f21a9db4b154c01d1170d4c>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.8.4-alpha.a6b828.0<br/>@codecademy/gamut@12.2.3-alpha.a6b828.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f21ee6763c7279a243f8100--gamut-preview.netlify.app>https://5f21ee6763c7279a243f8100--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f21ee6763c7279a243f8100>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.8.4-alpha.13797e.0<br/>@codecademy/gamut@12.2.3-alpha.13797e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f289b7bdc0ccbe3c032c569--gamut-preview.netlify.app>https://5f289b7bdc0ccbe3c032c569--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f289b7bdc0ccbe3c032c569>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f2c2a37bd70bb28e8cea67e--gamut-preview.netlify.app>https://5f2c2a37bd70bb28e8cea67e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f2c2a37bd70bb28e8cea67e>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.8.6-alpha.49a063.0<br/>@codecademy/gamut@12.2.5-alpha.49a063.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f2c6d9dc82ad9012d055f16--gamut-preview.netlify.app>https://5f2c6d9dc82ad9012d055f16--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f2c6d9dc82ad9012d055f16>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.9.3-alpha.f49e8a.0<br/>@codecademy/gamut@12.3.3-alpha.f49e8a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f343de7797786fb66774f25--gamut-preview.netlify.app>https://5f343de7797786fb66774f25--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f343de7797786fb66774f25>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.9.3-alpha.a36787.0<br/>@codecademy/gamut@12.3.3-alpha.a36787.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.9.3-alpha.4984f0.0<br/>@codecademy/gamut@12.3.3-alpha.4984f0.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.9.3-alpha.8d352d.0<br/>@codecademy/gamut@12.3.3-alpha.8d352d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f34437011d03b92a948c01f--gamut-preview.netlify.app>https://5f34437011d03b92a948c01f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f34437011d03b92a948c01f>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f3443c462e7f40f8e51c9cd--gamut-preview.netlify.app>https://5f3443c462e7f40f8e51c9cd--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f3443c462e7f40f8e51c9cd>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f3443d225bb64491e9e4787--gamut-preview.netlify.app>https://5f3443d225bb64491e9e4787--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f3443d225bb64491e9e4787>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.9.3-alpha.f4dbec.0<br/>@codecademy/gamut@12.3.3-alpha.f4dbec.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f34443862e7f4f14151cc21--gamut-preview.netlify.app>https://5f34443862e7f4f14151cc21--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f34443862e7f4f14151cc21>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.9.5-alpha.2ef526.0<br/>@codecademy/gamut@12.3.5-alpha.2ef526.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f3adc173b025ade5794fa3e--gamut-preview.netlify.app>https://5f3adc173b025ade5794fa3e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f3adc173b025ade5794fa3e>Deploy Logs</a></p>

**Pull request merged**